/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

import "google/protobuf/struct.proto";
import "fedlearner_webconsole/proto/common/extension.proto";

package fedlearner_webconsole.proto.testing;

enum Tenum {
  UNSPECIFIED = 0;
  E1 = 1;
  E2 = 2;
}

message PrivateInfo {
  string pii = 1 [(fedlearner_webconsole.proto.secret) = true];
  string non_pii = 2;
}

message RichMessage {
  string field1 = 1;
  int32 field2 = 2 [(fedlearner_webconsole.proto.secret) = true];
  repeated PrivateInfo infos = 3;
  PrivateInfo pinfo = 4;
  map<string, PrivateInfo> pinfo_map = 5;
  map<string, string> pstring_map = 6 [(fedlearner_webconsole.proto.secret) = true];
  repeated string pstring_list = 7 [(fedlearner_webconsole.proto.secret) = true];
}

message Tdata {
  int64 id = 1;
  repeated uint64 projects = 2;
  map<int32, google.protobuf.Value> mappers = 3;
  Tenum tt = 4;
}

message Int64Message {
  int64 id = 1;
  repeated Tdata data = 2;
  string uuid = 3;
  int32 project_id = 4;
}

message StructWrapper {
  google.protobuf.Value typed_value = 1;
  google.protobuf.Struct struct = 2;
}
